File: client/src/components/navbar.ts
import m from "mithril";

export class ComponentNavbar implements m.ClassComponent {
    view(vnode: m.CVnode) {
        return [
            m("div", { "nav": "primary", "flex": "grow", "bg": "black" }, [
                m("div", { "nav-section": "grow" }, [
                    m("div", { "nav-item": "brand" }, [
                        m(m.route.Link, { "href": "/home/1" }, "stry"),
                    ]),
                ]),
                m("div", { "nav-section": true }, [
                    m("div", { "nav-item": true }, [
                        m(m.route.Link, { "href": "/list/author" }, "register"),
                    ]),
                    m("div", { "nav-item": true }, [
                        m(m.route.Link, { "href": "/list/author" }, "login"),
                    ]),
                    m("input", { "nav-item": true, "type": "search", "placeholder": "search" }),
                ]),
            ]),
            m("div", { "nav": "secondary", "flex": "grow", "bg": "blue" }, [
                m("div", { "nav-section": "grow" }, [
                    m("div", { "nav-item": true }, [
                        m(m.route.Link, { "href": "/list/author" }, "authors"),
                    ]),
                    m("div", { "nav-item": true }, [
                        m(m.route.Link, { "href": "/list/origin" }, "origins"),
                    ]),
                    m("div", { "nav-item": true }, [
                        m(m.route.Link, { "href": "/list/warning" }, "warnings"),
                    ]),
                    m("div", { "nav-item": true }, [
                        m(m.route.Link, { "href": "/list/pairing" }, "pairings"),
                    ]),
                    m("div", { "nav-item": true }, [
                        m(m.route.Link, { "href": "/list/character" }, "characters"),
                    ]),
                    m("div", { "nav-item": true }, [
                        m(m.route.Link, { "href": "/list/tag" }, "tags"),
                    ]),
                ]),
            ]),
        ];
    }
}

File: client/src/components/navbar.rs
use {
    crate::{Message, Model},
    seed::{a, attrs, div, input, nav, prelude::*},
};

pub fn view(model: &Model) -> Vec<Node<Message>> {
    vec![
        nav![
            attrs! { "nav" => "primary", "flex" => "grow", "bg" => if model.debug { "red-500" } else { "black-800" } },
            div![
                attrs! { "nav-section" => "grow" },
                div![
                    attrs! { "nav-item" => "brand" },
                    a![attrs! { "href" => "#!/home/1" }, "stry"],
                ],
            ],
            div![
                attrs! { "nav-section" => true },
                input![
                    attrs! { "nav-item" => true, "name" => "search", "type" => "search", "placeholder"=> "search", "l-bg" => "white-200", "d-bg" => "black-700" },
                ],
            ],
        ],
        nav![
            attrs! { "nav" => "secondary", "flex" => "grow", "bg" => "blue-500" },
            div![
                attrs! { "nav-section" => "grow" },
                div![
                    attrs! { "nav-item" => true },
                    a![attrs! { "href" => "#!/list/authors/1" }, "authors"]
                ],
                div![
                    attrs! { "nav-item" => true },
                    a![attrs! { "href" => "#!/list/origins/1" }, "origins"]
                ],
                div![
                    attrs! { "nav-item" => true },
                    a![attrs! { "href" => "#!/list/warnings/1" }, "warnings"]
                ],
                div![
                    attrs! { "nav-item" => true },
                    a![attrs! { "href" => "#!/list/pairings/1" }, "pairings"]
                ],
                div![
                    attrs! { "nav-item" => true },
                    a![attrs! { "href" => "#!/list/characters/1" }, "characters"]
                ],
                div![
                    attrs! { "nav-item" => true },
                    a![attrs! { "href" => "#!/list/tags/1" }, "tags"]
                ]
            ]
        ],
    ]
}
